packages/airnode-deployer/src/infrastructure/aws.test.ts

```diff
@@ -0,0 +1,315 @@
+    awsListBucketsSpy.mockImplementation(mockPromise(() => listBucketsResponse));
+
+    await expect(getAirnodeBucket(cloudProvider)).rejects.toThrow(
+      new Error(`Multiple Airnode buckets found, stopping. Buckets: ${JSON.stringify(listBucketsResponse.Buckets)}`)
+    );
+    expect(awsConfigUpdateSpy).toHaveBeenCalledWith({ region: cloudProvider.region });
+  });
+});
+
+describe('createAirnodeBucket', () => {
+  beforeEach(() => {
+    awsCreateBucketSpy.mockImplementation(mockPromise(() => {}));
+    awsPutBucketEncryptionSpy.mockImplementation(mockPromise(() => {}));
+    awsPutPublicAccessBlockSpy.mockImplementation(mockPromise(() => {}));
+    generateBucketNameSpy.mockImplementation(() => bucketName);
+  });
+
+  it('creates S3 Airnode bucket', async () => {
+    await createAirnodeBucket(cloudProvider);
+
+    expect(awsCreateBucketSpy).toHaveBeenCalledWith({ Bucket: bucketName });
+    expect(awsPutBucketEncryptionSpy).toHaveBeenCalledWith({
+      Bucket: bucketName,
+      ServerSideEncryptionConfiguration: {
+        Rules: [{ ApplyServerSideEncryptionByDefault: { SSEAlgorithm: 'AES256' }, BucketKeyEnabled: true }],
+      },
+    });
+    expect(awsPutPublicAccessBlockSpy).toHaveBeenCalledWith({
+      Bucket: bucketName,
+      PublicAccessBlockConfiguration: {
+        BlockPublicAcls: true,
+        BlockPublicPolicy: true,
+        IgnorePublicAcls: true,
+        RestrictPublicBuckets: true,
+      },
+    });
+    expect(awsConfigUpdateSpy).toHaveBeenCalledWith({ region: cloudProvider.region });
+  });
+
+  it(`throws an error if can't create a bucket`, async () => {
+    awsCreateBucketSpy.mockImplementation(mockPromise(jest.fn().mockRejectedValue(s3Error)));
+
+    await expect(createAirnodeBucket(cloudProvider)).rejects.toThrow(
+      new Error(`Failed to create an S3 bucket: Error: ${s3ErrorMessage}`)
+    );
+    expect(awsConfigUpdateSpy).toHaveBeenCalledWith({ region: cloudProvider.region });
+  });
+
+  it(`throws an error if can't set enctyprion`, async () => {
+    awsPutBucketEncryptionSpy.mockImplementation(mockPromise(jest.fn().mockRejectedValue(s3Error)));
+
+    await expect(createAirnodeBucket(cloudProvider)).rejects.toThrow(
+      new Error(`Failed to enable encryption for bucket '${bucketName}': Error: ${s3ErrorMessage}`)
+    );
+    expect(awsConfigUpdateSpy).toHaveBeenCalledWith({ region: cloudProvider.region });
+  });
+
+  it(`throws an error if can't set public access`, async () => {
+    awsPutPublicAccessBlockSpy.mockImplementation(mockPromise(jest.fn().mockRejectedValue(s3Error)));
+
+    await expect(createAirnodeBucket(cloudProvider)).rejects.toThrow(
+      new Error(`Failed to setup a public access block for bucket '${bucketName}': Error: ${s3ErrorMessage}`)
+    );
+    expect(awsConfigUpdateSpy).toHaveBeenCalledWith({ region: cloudProvider.region });
+  });
+});
+
+describe('getBucketDirectoryStructure', () => {
+  it('returns bcuket directory structure', async () => {
```

Comment:
```suggestion
  it('returns bucket directory structure', async () => {
```

https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker